### PR TITLE
Fix tests build on MacOS

### PR DESCRIPTION
**Pull request by metalicjames**
This PR simplifies the googletest depedency finding in CMake, and removes the gmock dependency which isn't used. This fixes building the tests for me on MacOS, where it insisted on using the shared libraries for google test, even when only static libraries were installed.

### CODE REVIEW

**Comment by HalosGhost** [comment]
@metalicjames small merge conflict to fix.

**Comment by metalicjames** [comment]
@HalosGhost fixed!

**Comment by HalosGhost** [comment]
CC @AlexRamRam and @jonwiggins. I'll review this, but I don't have an apple machine available to test on. Would love to have a second set of eyes.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,6 @@
 project(tests)
 
-find_package(GTest REQUIRED)
-include(../3rdparty/FindGMock.cmake)
-find_package(Threads)
```
Comment (HalosGhost) [inline tests/CMakeLists.txt:5]: Is the removal of `find_package(Threads)` a part of the fix, or is this a separate fix/accidental-removal?

**Comment by metalicjames** [comment]
Without this change I get the following error when trying to run test binaries on MacOS.

```
dyld[89304]: Library not loaded: @rpath/libgtest_main.1.10.0.dylib
  Referenced from: /Users/james/workspace/cbdc-universe0/build/tests/integration/run_integration_tests
  Reason: tried: '/usr/local/lib/libgtest_main.1.10.0.dylib' (no such file), '/usr/lib/libgtest_main.1.10.0.dylib' (no such file)
```

**Review by HalosGhost — APPROVED** [review]
Solid concept-ACK. Both for the simplification, and also for fixing the tests on macOS more generally.

I have only one inline question, but I would like a Tested ACK / review from at least one person who has Apple hardware to verify.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (metalicjames) [reply tests/CMakeLists.txt:5]: It's not really a fix, just an unneeded line given it's already specified in a higher level `CMakeLists.txt` (here: https://github.com/mit-dci/opencbdc-tx/blob/trunk/CMakeLists.txt#L25).

I can move this removal to a separate commit if you'd prefer that.

Comment (HalosGhost) [reply tests/CMakeLists.txt:5]: No, that's just fine to be in here. I just didn't know if it was intentional. 👍

Comment (AlexRamRam) [reply tests/CMakeLists.txt:5]: > CC @AlexRamRam and @jonwiggins. I'll review this, but I don't have an apple machine available to test on. Would love to have a second set of eyes.

Other than #139, I am able to build and run the integration test on my Mac M1. Am I missing a part of the story?

**Comment by AlexRamRam** [comment]
> CC @AlexRamRam and @jonwiggins. I'll review this, but I don't have an apple machine available to test on. Would love to have a second set of eyes.

Other than https://github.com/mit-dci/opencbdc-tx/issues/139, I am able to build and run the integration test on my Mac M1. Am I missing a part of the story?

**Comment by jonwiggins** [comment]
This seems to introduce a breaking change for me on my M1 Mac.

On this branch if I run the configure script and then: 
```bash
opencbdc-tx-1> ./scripts/build.sh    
Building...
-- The C compiler identification is AppleClang 13.1.6.13160021
-- The CXX compiler identification is AppleClang 13.1.6.13160021
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test W_SHADOW_ALL
-- Performing Test W_SHADOW_ALL - Success
-- Performing Test W_NEWLINE_EOF
-- Performing Test W_NEWLINE_EOF - Success
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
Disabling Run Time Type Information (RTTI) features.
-- Configuring done
-- Generating done
-- Build files have been written to: {}/opencbdc-tx-1/build
[  2%] Building CXX object src/util/network/CMakeFiles/network.dir/connection_manager.o

{truncated normal looking output}

[ 82%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/atomizer_test.o
In file included from {}/opencbdc-tx-1/tests/integration/atomizer_end_to_end_test.cpp:14:
In file included from {}/opencbdc-tx-1/tests/./util.hpp:18:
/opt/homebrew/include/gtest/gtest.h:1358:11: error: comparison of integers of different signs: 'const unsigned long' and 'const int' [-Werror,-Wsign-compare]
  if (lhs == rhs) {
      ~~~ ^  ~~~
/opt/homebrew/include/gtest/gtest.h:1377:12: note: in instantiation of function template specialization 'testing::internal::CmpHelperEQ<unsigned long, int>' requested here
    return CmpHelperEQ(lhs_expression, rhs_expression, lhs, rhs);
           ^
{}/opencbdc-tx-1/tests/integration/atomizer_end_to_end_test.cpp:223:5: note: in instantiation of function template specialization 'testing::internal::EqHelper::Compare<unsigned long, int, nullptr>' requested here
    ASSERT_EQ(res_uhs_states.size(), 4);
    ^
/opt/homebrew/include/gtest/gtest.h:1877:31: note: expanded from macro 'ASSERT_EQ'
#define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
                              ^
/opt/homebrew/include/gtest/gtest.h:1861:54: note: expanded from macro 'GTEST_ASSERT_EQ'
  ASSERT_PRED_FORMAT2(::testing::internal::EqHelper::Compare, val1, val2)
                                                     ^
1 error generated.
make[2]: *** [tests/integration/CMakeFiles/run_integration_tests.dir/atomizer_end_to_end_test.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 83%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/buffer_test.o
[ 83%] Built target locking-shardd
[ 83%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/common/hash_test.o
[ 83%] Built target client-cli
[ 84%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/config_test.o
[ 84%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/coordinator/messages_test.o
[ 84%] Built target coordinatord
[ 85%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/locking_shard/format_test.o
[ 85%] Linking CXX executable twophase-gen
[ 86%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/network_test.o
[ 86%] Built target twophase-gen
[ 86%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/message_test.o
[ 87%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/raft_test.o
[ 87%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/rpc/tcp_test.o
[ 88%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/sentinel_2pc/controller_test.o
[ 89%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/serialization_test.o
[ 89%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/serialization/format_test.o
[ 90%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/shard_test.o
make[1]: *** [tests/integration/CMakeFiles/run_integration_tests.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 90%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/socket_test.o
[ 91%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/serialization/stream_serializer_test.o
[ 91%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/transaction_test.o
[ 92%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/twophase_test.o
[ 93%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/validation_test.o
[ 93%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/wallet_test.o
[ 94%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/watchtower/block_cache_test.o
[ 94%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/watchtower/error_cache_test.o
[ 95%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/watchtower/tx_error_messages_test.o
[ 96%] Building CXX object tests/unit/CMakeFiles/run_unit_tests.dir/watchtower/watchtower_test.o
[ 96%] Linking CXX executable run_unit_tests
[ 96%] Built target run_unit_tests
make: *** [all] Error 2
```

On `opencbdc-tx:trunk` I can run
```bash
opencbdc-tx> ./scripts/build.sh    
Building...
Disabling Run Time Type Information (RTTI) features.
-- Configuring done
-- Generating done
-- Build files have been written to: {}/opencbdc-tx/build
Consolidate compiler generated dependencies of target secp256k1_genctx
Consolidate compiler generated dependencies of target crypto
Consolidate compiler generated dependencies of target bech32
Consolidate compiler generated dependencies of target archiver
Consolidate compiler generated dependencies of target transaction
Consolidate compiler generated dependencies of target atomizer
Consolidate compiler generated dependencies of target watchtower
Consolidate compiler generated dependencies of target serialization
Consolidate compiler generated dependencies of target network
Consolidate compiler generated dependencies of target common
{truncated normal looking output}
[ 90%] Linking CXX executable run_integration_tests
[ 91%] Built target run_integration_tests
[ 91%] Linking CXX executable run_unit_tests
[100%] Built target run_unit_tests
```

Is it possible there is an additional gtest library that needs to be added to `./scripts/configure.sh`?

**Comment by metalicjames** [comment]
@jonwiggins I pushed [91435cb](https://github.com/mit-dci/opencbdc-tx/pull/143/commits/91435cb13149a5ca13bd00cc00c74b853ced56bd) which should fix the compiler error you're getting.

**Comment by jonwiggins** [comment]
I get the same error here: https://github.com/mit-dci/opencbdc-tx/blob/ca0af9e132a47744e0e6745815de94a027823151/tests/integration/atomizer_end_to_end_test.cpp#L272

Making the same change to `4UL` works.

After that change I think everything looks good.

**Comment by metalicjames** [comment]
@jonwiggins I updated the commit to add the change.

**Review by jonwiggins — APPROVED** [review]
Looks great to me!

**Review by HalosGhost — APPROVED** [review]
I took another look, and this fix still looks solid. With @jonwiggins's sign-off, I'm happy to merge.

**Pull request merged**